### PR TITLE
API spec v0.2.0

### PR DESCRIPTION
**Pull request by sherifnada**
## What
* Attaches a default DockerRepository and imageVersion to source/destination
* keys SourceSpec/DestinationSpec on Source/Destination instead of having its own primary key
* Adds the ability to create sources, and update a source or destination's default imageVersion

in the future, specs will be retrieved based on a (source/dest ID, source/dest version) tuple or a dedicated ID representing that tuple. But right now since there is no explicit versioning registry, we always use the source/destination's default image version. 

If this looks good I'll go ahead and start editing the "inner" schemas behind the scenes

### CODE REVIEW

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -1458,4 +1512,4 @@ components:
     InvalidInput:
       description: Invalid Input
 security:
-  - {}
+  - { }
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1515]: ?

File: airbyte-api/src/main/openapi/config.yaml [hunk 2]
```diff
@@ -121,6 +121,48 @@ paths:
           description: Workspace not found
         "422":
           $ref: "#/components/responses/InvalidInput"
+  /v1/sources/create:
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:124]: what is calling this?

File: airbyte-api/src/main/openapi/config.yaml [hunk 3]
```diff
@@ -895,21 +967,15 @@ components:
           items:
             $ref: "#/components/schemas/SourceRead"
     # SOURCE SPECIFICATION
-    SourceSpecificationId:
-      type: string
-      format: uuid
     SourceSpecification:
       description: The specification for what values are required to configure the source.
       example: { user: { type: string } }
     SourceSpecificationRead:
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:973]: will this now read from the spec method on each docker image?

**Review by cgardens — COMMENTED** [review]
> Adds the ability to create sources, and update a source's default imageVersion

why just source and not destination?

Just commenting, because I feel like I'm missing an important piece here, which is what will be calling `create` / `update`. Definitely don't need you to implement it yet, but a few sentences explaining what you have in mind would be helpful so I have a better idea of how this will be used.

File: airbyte-api/src/main/openapi/config.yaml (hunk 3 above)
Comment (sherifnada) [reply airbyte-api/src/main/openapi/config.yaml:973]: That is the idea, but it doesn't have to happen in strict lockstep with this change.

**Comment by sherifnada** [comment]
Screen Shot 2020-10-02 at 12 02 05 PM

@cgardens the create endpoint will be called from this screen in the current mockups

**Comment by sherifnada** [comment]
> why just source and not destination?

We'll be able to change a destination's version but not add a new destination since we only support one destination rn.

**Comment by sherifnada** [comment]
initial PR did not have destination update endpoint, added now.

File: airbyte-api/src/main/openapi/config.yaml (hunk 1 above)
Comment (sherifnada) [reply airbyte-api/src/main/openapi/config.yaml:1554]: ij formatting, removed

**Review by cgardens — APPROVED** [review]

File: airbyte-api/src/main/openapi/config.yaml (hunk 2 above)
Comment (michel-tricot) [reply airbyte-api/src/main/openapi/config.yaml:124]: Just to make sure we are fully aligned on the terminology:

does it mean we will call source-connector: source?
does it mean we will call destination-connector: destination?

Challenging it: why don't we call it connector (with an enum type to distinguish between source and destination)?

File: airbyte-api/src/main/openapi/config.yaml [hunk 4]
```diff
@@ -875,16 +940,46 @@ components:
       properties:
         sourceId:
           $ref: "#/components/schemas/SourceId"
+    SourceCreate:
+      type: object
+      required:
+        - name
+        - defaultDockerRepository
```
Comment (michel-tricot) [inline airbyte-api/src/main/openapi/config.yaml:947]: why is it called default?

Comment (sherifnada) [reply airbyte-api/src/main/openapi/config.yaml:947]: future-proofing to allow specific connector instances to diverge from the global default version

**Comment by sherifnada** [comment]
Superseded by #528 and #529

**Pull request closed without merging**